app/controllers/users_controller.rb

```diff
@@ -67,6 +67,22 @@ def edit
     @followers = Follower.where(user: params[:id]).includes(:following)
   end
 
+  def delete
+    @user = User.find(params[:id])
+  end
+
+  def destroy
+    @user = User.find(params[:id])
+    unless @user.authenticate(params[:oldpassword])
+      @user.errors[:oldpassword][0] = 'Invalid password.'
+      render 'delete'
+      return
+    end
+    UserFeedback.create(title: 'Account Deleted', body: params[:body]) if params[:body].present?
+    @user.destroy
+    redirect_to root_url
+  end
```

Comment:
I think I mentioned this at some point (and I hope it's not too much hassle to manage), but I want to do the sort of "second thoughts" optioning of deletion.
What I mean by this is that when a user indicates they want to delete their account, it would be better to have a flag turned on (something like days_to_deletion, and set it to fourteen days), and have that decrement each rollover until it hits zero, **then delete it.**
The reason behind this thinking is that in many, many instances as an admin, it appears that people often either regret deleting their account, or reconsider after 48 hours to a week.
The second reason is while there aren't any glaring security issues with DAD as of this moment, as far as I can tell, if someone inadvertently could gain access to their account, or even just the delete feature, I'd prefer it to be a lot less easy for them to delete someone's account. I'm a bit paranoid about this: I literally discovered two years after DAD's implementation that you could edit other peoples' submissions. Not great on my part.

Reply:
Made a big-ass button

![image](https://user-images.githubusercontent.com/36219436/144742454-79607c6a-3a9a-4def-ad92-10ac87c508dd.png)